stdlib/zipfile/__init__.pyi

```diff
@@ -225,8 +230,17 @@ class ZipFile:
     def getinfo(self, name: str) -> ZipInfo: ...
     def infolist(self) -> list[ZipInfo]: ...
     def namelist(self) -> list[str]: ...
+    @overload
+    def open(
+        self, name: str | ZipInfo, mode: Literal["r"] = "r", pwd: bytes | None = None, *, force_zip64: bool = False
+    ) -> ZipExtFile: ...
+    @overload
+    def open(
+        self, name: str | ZipInfo, mode: Literal["w"] = ..., pwd: bytes | None = None, *, force_zip64: bool = False
```

Comment:
This would imply that this overload would also apply if no `mode` argument is given, which obviously isn't true. Also, since `pwd` is only allowed when reading files:

```suggestion
    @overload
    def open(
        self, name: str | ZipInfo, mode: Literal["w"], pwd: None = None, *, force_zip64: bool = False
```